app/src/main/java/com/pps/globant/fittracker/mvp/presenter/InstagramLoginPresenter.java

```diff
@@ -0,0 +1,85 @@
+package com.pps.globant.fittracker.mvp.presenter;
+
+import android.app.Dialog;
+import android.webkit.WebView;
+
+import com.pps.globant.fittracker.MainActivity;
+import com.pps.globant.fittracker.mvp.model.InstagramLoginModel;
+import com.pps.globant.fittracker.mvp.view.IntagramLoginView;
+import com.pps.globant.fittracker.utils.BusProvider;
+import com.pps.globant.fittracker.utils.MyWVClient;
+import com.squareup.otto.Bus;
+import com.squareup.otto.Subscribe;
+
+public class InstagramLoginPresenter {
+    private final static String CODE = "code";
+    private final static String ERROR = "error";
+    private static final String EQUAL_SIGN = "=";
+    private final InstagramLoginModel model;
+    private final IntagramLoginView view;
+    public Dialog dialog;
+    public MainActivity activity;
+    public Bus bus;
+
+
+    public InstagramLoginPresenter(Bus bus) {
+        this.bus = bus;
+        BusProvider.register(this);
+        model = new InstagramLoginModel(bus);
+        view = new IntagramLoginView();
+    }
+
+    public void igButtonLoginClick(MainActivity activity) {
+        this.activity = activity;
+        dialog = new Dialog(activity);
+        WebView webView = new WebView(activity);
```

Comment:
Why do you need a WebView?

Reply:
i will commit in this pull request again with the webView, having fixed all the others reviews, to dont lose more time with the login and can start with the API but however i will study the implementation with an intent opening a browser like you recommend for another PR